plugins/fixed-edges/src/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/**
+ * @fileoverview Plugin overview.
+ */
+
+import Blockly from 'blockly/core';
+
+let fixedEdges = {
+  top: true,
+  left: true,
+};
+
+/**
+ * The manager for a workspace with fixed edges.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to calculate metrics
+ *     for.
+ * @implements {Blockly.IMetricsManager}
+ * @constructor
+ */
+export class FixedEdgesMetricsManager extends Blockly.MetricsManager {
+  /**
+   * Constructor for ...
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
+   *     be added to.
+   */
+  constructor(workspace) {
+    super(workspace);
+  }
+
+  /**
+   * Sets which edges are fixed.
+   * @param {{
+   *   top: (boolean|undefined),
+   *   bottom: (boolean|undefined),
+   *   left: (boolean|undefined),
+   *   right: (boolean|undefined)
+   * }} newFixedEdges The edges to set as fixed.
+   */
+  static setFixedEdges(newFixedEdges) {
```

Comment:
I'd like feedback on this method.
I'm not confident with this choice of param type.
My intention is that you can set top or bottom or neither as "fixed"/"contant" AND left or right or neither as "fixed"/"contant".

Reply:
updated wording throughout